### PR TITLE
Add official support for Django 2.2.

### PR DESCRIPTION
**Pull request by joehybird**

### CODE REVIEW

**Comment by joehybird** [comment]
Why coverage from 95% (master) to 97% (this branch) is show as decreased ?

File: tests/tests.py [hunk 1]
```diff
@@ -35,6 +40,7 @@ def test_data_integrity(self):
             # the same data, even thought it's stored differently in the DB.
             self.assertEqual(value, model_test.pickle_field)
             self.assertEqual(value, model_test.compressed_pickle_field)
+            self.assertEqual(None, model_test.nullable_pickle_field)
```
Comment (charettes) [inline tests/tests.py]:
```suggestion
            self.assertIsNone(model_test.nullable_pickle_field)
```

File: picklefield/fields.py [hunk 2]
```diff
@@ -170,12 +169,8 @@ def pre_save(self, model_instance, add):
...
         return wrap_conflictual_object(value)
 
-    if django.VERSION < (2, 0):
-        def from_db_value(self, value, expression, connection, context):
-            return self.to_python(value)
-    else:
-        def from_db_value(self, value, expression, connection):
-            return self.to_python(value)
+    def from_db_value(self, value, expression, connection, context=None):
+        return self.to_python(value)
```
Comment (charettes) [inline picklefield/fields.py]: This is still necessary unless you drop support for Django 1.11.

File: tox.ini [hunk 3]
```diff
@@ -4,28 +4,32 @@ args_are_paths = false
 envlist =
     flake8,
     isort,
-    py27-1.11,
-    py34-{1.11,2.0},
-    py35-{1.11,2.0,2.1,master},
-    py36-{1.11,2.0,2.1,master},
-    py37-{1.11,2.0,2.1,master}
+    py{27,36,37,38}-1.11,
```
Comment (charettes) [inline tox.ini]: 1.11 doesn't support 3.7 and 3.8, see https://djangoci.com/job/django-1.11/

File: tox.ini [hunk 4]
```diff
@@ -4,28 +4,32 @@ args_are_paths = false
...
+    py{36,37,38}-{2.0,2.1,2.2,master}
```
Comment (charettes) [inline tox.ini]: 2.0 and 2.1 are unsupported

2.2 supports 3.5 as well https://djangoci.com/job/django-2.2/

File: tox.ini [hunk 5]
```diff
@@ -4,28 +4,32 @@ args_are_paths = false
...
+2.7 = py27
+3.6 = py36
+3.7 = py37
+3.8 = py38
 
 [testenv]
 basepython =
     py27: python2.7
-    py34: python3.4
-    py35: python3.5
```
Comment (charettes) [inline tox.ini]: The 2.2 LTS supports 3.5

File: tests/tests.py [hunk 6]
```diff
@@ -181,6 +187,21 @@ def test_empty_strings_not_allowed(self):
...
+        model.save()
+
+        self.assertEqual(
+            {'foo': 'bar'}, MinimalTestingModel.objects.get(pk=model.pk).pickle_field
+        )
+
+        with patch('picklefield.fields.dbsafe_decode', mock_decode_error):
+            encoded_value = dbsafe_encode({'foo': 'bar'})
+            self.assertEqual(encoded_value, MinimalTestingModel.objects.get(pk=model.pk).pickle_field)
+
```
Comment (charettes) [inline tests/tests.py]: What does this patch have to do with Django 2.2 support?

Comment (joehybird) [reply tests/tests.py]: Nothing, just coverage improvement :smile:

**Comment by joehybird** [comment]
Ok I will split it in different commits (should be easy... I hope so...)

File: picklefield/fields.py (hunk 2 above)
Comment (joehybird) [reply picklefield/fields.py]: It works in all versions with full coverage, but the django 3.0 warning is back. Pretty strange because master (3.1) is ok.

**Comment by joehybird** [comment]
Ok fixed :)

**Comment by charettes** [comment]
Awesome thanks for everything!

**Comment by joehybird** [comment]
You can thank me with new release :pray:

**Comment by charettes** [comment]
Et voilà https://pypi.org/project/django-picklefield/

**Pull request merged**